### ISSUE TITLE
synchronize module, missing quotes on rsync arguments

### ISSUE THREAD

**Issue by Kostafun** [ansible/ansible#23887]
##### ISSUE TYPE

 - Bug Report

##### COMPONENT NAME

- synchronize
##### ANSIBLE VERSION

```
2.3.0.0
```

##### CONFIGURATION

- ansible.cfg
```
[ssh_connection]
ssh_args = -o ControlMaster=auto -o ControlPersist=600s
control_path = %(directory)s/%%h-%%r
pipelining = True
```

##### OS / ENVIRONMENT

```
Linux 4.10.10-1-MANJARO
```
##### SUMMARY

Rsync synchronization does not work on 2.3.0.0, works on 2.2.2.0

##### STEPS TO REPRODUCE

```yaml
- name: Rsync upload to server
  synchronize: src={{ config_local_path }} dest={{ rails_release_path }} rsync_opts=--exclude-from='{{ config_local_path }}.gitignore' owner=no group=no
  become: yes
  become_user: "{{ user }}"
```

##### EXPECTED RESULTS

On version 2.2.2.0 works as expected, executing
```
/usr/bin/rsync --delay-updates -F --compress --archive --no-owner --no-group --rsh 'ssh  -S none -o StrictHostKeyChecking=no' --rsync-path=\"sudo rsync\" --exclude-from='/home/kostafun/Projects/domain.com/.gitignore' --out-format='<<CHANGED>>%i %n%L' \"/home/kostafun/Projects/domain.com/\" \"root@2.2.2.2:/home/bk/domain/releases/20170423080009\"
```

##### ACTUAL RESULTS

On 2.3.0.0 executing 
```
 /usr/bin/rsync --delay-updates -F --compress --archive --no-owner --no-group --rsh=/usr/bin/ssh -S none -o StrictHostKeyChecking=no --rsync-path=sudo rsync --exclude-from='/home/kostafun/Projects/domain.com/.gitignore' --out-format=<<CHANGED>>%i %n%L /home/kostafun/Projects/domain.com/ root@2.2.2.2:/home/bk/domain/releases/20170423070910
```
with the following results
```
rsync: failed to open exclude file '/home/kostafun/Projects/domain.com/.gitignore': No such file or directory (2)\nrsync error: error in file IO (code 11) at exclude.c(1178) [client=3.1.2]\n
```
Note no quotes --rsync-path=sudo rsync  --out-format=<<CHANGED>>%i %n%L 
Commit #23157 ?

Complete -vvv output:
```
Using module file /usr/lib/python2.7/site-packages/ansible/modules/files/synchronize.py
<2.2.2.2> ESTABLISH LOCAL CONNECTION FOR USER: kostafun
<2.2.2.2> EXEC /bin/sh -c '/usr/bin/python2 && sleep 0'
fatal: [2.2.2.2]: FAILED! => {
    "changed": false, 
    "cmd": "/usr/bin/rsync --delay-updates -F --compress --archive --no-owner --no-group --rsh=/usr/bin/ssh -S none -o StrictHostKeyChecking=no --rsync-path=sudo rsync --exclude-from='/home/kostafun/Projects/domain.com/.gitignore' --out-format=<<CHANGED>>%i %n%L /home/kostafun/Projects/domain.com/ root@2.2.2.2:/home/bk/domain/releases/20170423070910", 
    "failed": true, 
    "invocation": {
        "module_args": {
            "_local_rsync_path": "rsync", 
            "_substitute_controller": false, 
            "archive": true, 
            "checksum": false, 
            "compress": true, 
            "copy_links": false, 
            "delete": false, 
            "dest": "root@2.2.2.2:/home/bk/domain/releases/20170423070910", 
            "dest_port": null, 
            "dirs": false, 
            "existing_only": false, 
            "group": false, 
            "links": null, 
            "mode": "push", 
            "owner": false, 
            "partial": false, 
            "perms": null, 
            "private_key": null, 
            "recursive": null, 
            "rsync_opts": [
                "--exclude-from='/home/kostafun/Projects/domain.com/.gitignore'"
            ], 
            "rsync_path": "sudo rsync", 
            "rsync_timeout": 0, 
            "set_remote_user": true, 
            "src": "/home/kostafun/Projects/domain.com/", 
            "ssh_args": null, 
            "times": null, 
            "verify_host": false
        }
    }, 
    "msg": "rsync: failed to open exclude file '/home/kostafun/Projects/domain.com/.gitignore': No such file or directory (2)\nrsync error: error in file IO (code 11) at exclude.c(1178) [client=3.1.2]\n", 
    "rc": 11
}

```

**Comment by peteeckel** [comment]
I found out a similar issue with quoting in synchronize yesterday:

    - name: Update the mirrors (this may take a very long time)
      synchronize: >
          src={{ item.source }}
          dest=/data/mirrors/{{ item.destination }}
          archive=yes
          copy_links={{ item.copy_links }}
          mode=pull
          rsync_opts="{% for pattern in item.excludes %}--exclude='{{ pattern }}' {% endfor %}"
      with_items: "{{ synchronize_repositories }}"

The synchronize module actually works in this case, *but* removes the single quotes around the --exclude arguments. This has the effect of changing the command on the target system from

    /usr/bin/rsync [...] --exclude='*/aarch64/*' --exclude='*/ppc64/*' --exclude='*/ppc64le/*' [...] 

to

    /usr/bin/rsync [...] --exclude=*/aarch64/* --exclude=*/ppc64/* --exclude=*/ppc64le/* [...]

which renders the --exclude option ineffective.

**Comment by erwilan** [comment]
this is a duplicate of #23575

**Comment by alikins** [comment]
I think this may be resolved in devel/ and in the stable-2.3 branch via f7c9f44aabbabe371358672be2381f9a11073f96 and c16622b072203b1cbe59f937e3920741db8db6ef)

Could you test this with a build from the stable-2.3 branch? 

Also see https://github.com/ansible/ansible/issues/23575#issuecomment-296570940

**Comment by alikins** [comment]
ansibot: duplicate_of 23575
